packages/bsky/src/data-plane/bsync/index.ts

```diff
@@ -228,51 +228,45 @@ const handleSubjectActivitySubscriptionOperation = async (
 ) => {
   const { actorDid, key, method, payload } = req
 
-  const parsed: SubjectActivitySubscription = JSON.parse(
+  if (method === Method.DELETE) {
+    return db.db
+      .deleteFrom('activity_subscription')
+      .where('creator', '=', actorDid)
+      .where('key', '=', key)
+      .execute()
+  }
+
+  const parsed = jsonStringToLex(
     Buffer.from(payload).toString('utf8'),
-  )
-  const { activitySubscription } = parsed
-  const { post, reply } = activitySubscription
-  // We only keep a record if enabled.
-  const enabled = isActivitySubscriptionEnabled(activitySubscription)
+  ) as SubjectActivitySubscription
+  const {
+    subject,
+    activitySubscription: { post, reply },
+  } = parsed
 
   if (method === Method.CREATE) {
-    if (enabled) {
-      return db.db
-        .insertInto('activity_subscription')
-        .values({
-          creator: actorDid,
-          subjectDid: parsed.subject,
-          key,
-          createdAt: now,
-          indexedAt: now,
-          post,
-          reply,
-        })
-        .execute()
-    }
-    return
-  }
-
-  if (method === Method.UPDATE && enabled) {
     return db.db
-      .updateTable('activity_subscription')
-      .where('creator', '=', actorDid)
-      .where('subjectDid', '=', parsed.subject)
-      .where('key', '=', key)
-      .set({
+      .insertInto('activity_subscription')
+      .values({
+        creator: actorDid,
+        subjectDid: subject,
+        key,
+        indexedAt: now,
         post,
         reply,
       })
       .execute()
   }
 
-  // A DELETE, or an UPDATE with subscription disabled.
   return db.db
-    .deleteFrom('activity_subscription')
+    .updateTable('activity_subscription')
```

Comment:
Uniqueness constraint on (creator, subject) might fail.  Let's at least ensure the write to the stash doesn't fail for the caller.